src/nn/var_store.rs

```diff
@@ -265,6 +294,62 @@ impl<'a> Path<'a> {
         }
     }
 
+    /// Casts all float-like variables in a sub-path to half-precision (Half kind).
+    ///
+    /// Only the variable in the path sub-tree are cast to half-precision:
+    /// other VarStore variables are unaffected
+    pub fn half(&mut self) {
+        let path_root = self.path.join(SEP.to_string().as_str());
+        let mut variables = self.var_store.variables_.lock().unwrap();
+        for (variable_name, variable) in variables.named_variables.iter_mut() {
+            if variable_name.starts_with(&path_root) & variable.is_floating_point() {
+                *variable = variable.to_kind(Kind::Half);
+            }
+        }
+    }
```

Comment:
Would it be possible to have a function with a kind argument: it would avoid duplicating the code 4 times and also could be exposed to the end users if they want to use a kind argument.

Reply:
Thank you for the suggestion. I have added a new `to_kind` method exposed to the user that would allow to convert a path/var store to any `Kind`. I would rely on a different (private) `to_float_kind` for the implementation of `half`, `float`, `bfloat16` and `double` to only process float-like variables in the var store. This would be matching Pytorch's implementation at https://pytorch.org/docs/stable/_modules/torch/nn/modules/module.html#Module.half